### PR TITLE
 #805 implemented MkBout#update()

### PR DESCRIPTION
**Pull request by dmzaytsev**
for #805 
* implemented `MkBout#update()`
* implemented `TochBout` to change the last update attribute
* created unit tests
* removed redundant puzzle

### CODE REVIEW

**Comment by karato** [comment]
@dmzaytsev Thanks for your pull request, let me find someone who can review it

**Comment by karato** [comment]
@krzyk do this review

File: netbout-client/src/main/java/com/netbout/mock/MkBout.java [hunk 1]
```diff
@@ -101,13 +102,17 @@ public Date date() throws IOException {
 
     /**
      * {@inheritDoc}
-     * @todo #806:30min/DEV This method needs to be implemented
```
Comment (krzyk) [inline netbout-client/src/main/java/com/netbout/mock/MkBout.java]: @dmzaytsev this todo didn't produce an issue?

File: netbout-client/src/main/java/com/netbout/mock/MkBout.java [hunk 2]
```diff
@@ -101,13 +102,17 @@ public Date date() throws IOException {
 
     /**
      * {@inheritDoc}
```
Comment (krzyk) [inline netbout-client/src/main/java/com/netbout/mock/MkBout.java]: @dmzaytsev `{@inheritDoc}` is not required for overriden methods

File: netbout-client/src/main/java/com/netbout/mock/TouchBout.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * Touch bout.
+ *
+ * @author Dmitry Zaytsev (dmitr.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 2.23
+ */
+@Immutable
+@ToString
+@Loggable(Loggable.DEBUG)
+@EqualsAndHashCode(of = "bout")
```
Comment (krzyk) [inline netbout-client/src/main/java/com/netbout/mock/TouchBout.java]: @dmzaytsev `sql` param is missing

File: netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java [hunk 4]
```diff
@@ -69,4 +69,27 @@ public void uploadsAndDownloads() throws Exception {
...
+    @Test
+    public void changesUpdateAttribute() throws Exception {
+        final Bout bout = new MkBase().randomBout();
+        final Attachments attachments = bout.attachments();
+        final Long last = bout.updated().getTime();
+        final String name = "att-name";
+        attachments.create(name);
+        final Attachment attachment = attachments.get(name);
+        attachment.write(
+            IOUtils.toInputStream("<root/>", CharEncoding.UTF_8),
```
Comment (krzyk) [inline netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java]: @dmzaytsev it is better to use `StandardCharsets.UTF_8` from JDK

File: netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java [hunk 5]
```diff
@@ -69,4 +69,27 @@ public void uploadsAndDownloads() throws Exception {
...
+            "text/xml",
+            Long.toString(System.currentTimeMillis())
+        );
+        final Long pause = 100L;
```
Comment (krzyk) [inline netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java]: @dmzaytsev why a variable? you could use `Tv.HUNDRED` in the sleep below

File: netbout-client/src/test/java/com/netbout/mock/MkMessagesTest.java [hunk 6]
```diff
@@ -61,4 +61,20 @@ public void postsAndReadsMessages() throws Exception {
...
+     * MkMessages can change update attribut on Bout.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    public void changesUpdateAttribute() throws Exception {
+        final Bout bout = new MkBase().randomBout();
+        final Messages messages = bout.messages();
+        final Long last = bout.updated().getTime();
+        messages.post("hi");
+        final Long pause = 100L;
```
Comment (krzyk) [inline netbout-client/src/test/java/com/netbout/mock/MkMessagesTest.java]: @dmzaytsev similar thing here, why a variable?

File: netbout-client/src/test/java/com/netbout/mock/TouchBoutTest.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+     * TouchBout can change update attribute of the bout.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    public void changesUpdateAttribute() throws Exception {
+        final Sql sql = new H2Sql();
+        final Bout bout = new MkBase(sql).randomBout();
+        final Long last = bout.updated().getTime();
+        new TouchBout(sql, bout.number()).act();
+        final Long pause = 100L;
```
Comment (krzyk) [inline netbout-client/src/test/java/com/netbout/mock/TouchBoutTest.java]: @dmzaytsev and here

**Comment by krzyk** [comment]
@dmzaytsev please see my comments

File: netbout-client/src/main/java/com/netbout/mock/MkBout.java (hunk 1 above)
Comment (dmzaytsev) [inline netbout-client/src/main/java/com/netbout/mock/MkBout.java]: @krzyk ticket #805 has been created first  and then the puzzle #827 
I missed it
If i removed this todo I guess #827 will be closed 
no one assigned to that puzzle

File: netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java (hunk 4 above)
Comment (dmzaytsev) [inline netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java]: @krzyk you're right
I just copy/paste it from another test

File: netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java (hunk 5 above)
Comment (dmzaytsev) [inline netbout-client/src/test/java/com/netbout/mock/MkAttachmentsTest.java]: @krzyk `Tv.HUNDRED` is the same magic number as digits... :)

**Comment by dmzaytsev** [comment]
@krzyk done
please take a look

**Comment by krzyk** [comment]
@dmzaytsev looks good

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by krzyk** [comment]
@dmzaytsev I think you should assign yourself to #805

**Comment by dmzaytsev** [comment]
@krzyk sure, thank you!

**Comment by karato** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@karato Looks good!

**Comment by karato** [comment]
> @karato Looks good!

@elenavolokhova thanks!

**Comment by karato** [comment]
@krzyk since quality is good, 10 mins paid to @dmzaytsev (our architect), transaction `75634408`

**Comment by karato** [comment]
@krzyk added 10 mins to @elenavolokhova, for QA review, payment code is `75641544`... Thanks so much! Your account was topped up for **26 mins** (transaction ID is `75641549`, the task took 23 hours and 48 mins)... review comments (c=11) added as a bonus... added +26 to your rating, now it is equal to [+4269](http://www.netbout.com/b/35035?open=rating)

**Pull request merged**